chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/ComplaintEntity.java

```diff
@@ -0,0 +1,225 @@
+package gov.healthit.chpl.entity;
+
+import java.util.Date;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "complaint")
+public class ComplaintEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "complaint_id", nullable = false)
+    private Long id;
+
+    @Basic(optional = true)
+    @OneToOne(optional = true, fetch = FetchType.LAZY)
+    @JoinColumn(name = "certification_body_id", nullable = false, insertable = true, updatable = false)
+    private CertificationBodyEntity certificationBody;
+
+    @Basic(optional = true)
+    @OneToOne(optional = true, fetch = FetchType.LAZY)
+    @JoinColumn(name = "complaint_type_id", nullable = true, insertable = true, updatable = true)
```

Comment:
If this works, I guess it's fine, but normally in the entities when we are joining in another entity like this we have a @Column defined for the bigint foreign key value (complaint_type_id) and that column is allowed to be inserted and updated but this column that's going to hold the actual entity that gets joined in is not insertable/updatable. I think if you look at any of the Surveillance entities you will see a good example. I would be afraid this syntax would let the underlying ComplaintTypeEntity/ComplaintStatusTypeEntity/CertificationBodyEntity get updated.

Reply:
I'll make this change, and I have seen it throughout the code.  But it feel like it's hack.  I have never heard or seen this done other places or in code examples on the web.